Meta/CMake/wasm_spec_tests.cmake

```diff
@@ -0,0 +1,33 @@
+option(INCLUDE_WASM_SPEC_TESTS "Download and include the WebAssembly spec testsuite" OFF)
+
+if(INCLUDE_WASM_SPEC_TESTS)
+    set(SOURCE_DIR ${CMAKE_SOURCE_DIR})
+    if (CMAKE_SOURCE_DIR MATCHES ".*/Lagom")
+        set(SOURCE_DIR ${SOURCE_DIR}/../..)
+    endif()
+    set(WASM_SPEC_TEST_GZ_URL https://github.com/WebAssembly/testsuite/archive/refs/heads/master.tar.gz)
+    set(WASM_SPEC_TEST_GZ_PATH ${CMAKE_BINARY_DIR}/wasm-spec-testsuite.tar.gz)
+    set(WASM_SPEC_TEST_TAR_PATH ${CMAKE_BINARY_DIR}/wasm-spec-testsuite.tar)
+    set(WASM_SPEC_TEST_PATH ${SOURCE_DIR}/Userland/Libraries/LibWasm/Tests/Fixtures/SpecTests)
```

Comment:
This looks suspiciously like a [`FetchContent`](https://cmake.org/cmake/help/latest/module/FetchContent.html) but I don't know enough about CMake to know the magic incantation to get the same result.

Reply:
"this" is a bunch of variable definitions :confused:
Here's what it does:
- download the tests
- extract them to a temporary directory
- generate a bunch of `foo.js` and `foo.wasm` files (via `generate-libwasm-spec-test.py`) and put *those* into the Tests dir of LibWasm
- get rid of the downloaded stuff

`FetchContent` seems to be a more advanced version of the download thing, and I don't see a preprocess step in its manual page

Reply:
Hmm yeah. by "this" I meant the entire file. The download and extract part is what could be covered by FetchContent, if we wanted. It would also ensure that things always end up in a child of the build dir. Build/lagom/_deps/wasm-spec/ for example. But this approach works as-is.

Reply:
Hmmm
we could switch the PCI id stuff to use this too, though I'm not sure what this offers over `file(DOWNLOAD ...)`
I'm not a knows-cmake guy, I just look at the manual...and the manual doesn't say when to use this and when to use that...
so monkey see, monkey copy-paste lol